controllers/predicate/predicate.go

```diff
@@ -179,8 +149,9 @@ func EtcdReconciliationFinished(ignoreOperationAnnotation bool) predicate.Predic
 	}
 }
 
-// IsSnapshotLease is a predicate that is `true` if the passed lease object is a snapshot lease.
-func IsSnapshotLease() predicate.Predicate {
+// SnapshotLeaseChanged is a predicate that is `true` if the passed lease object is a snapshot lease and if the lease
+// object's holderIdentity is updated.
+func SnapshotLeaseChanged() predicate.Predicate {
```

Comment:
The lease holder identity currently holds the revision number. What you compare is only the revision. Why not just name it - `SnapshotRevisionChanged` which is semantically correct name for the function?

Reply:
I think it is easier for anyone going through the code for the first time to understand that this function is dealing with Snapshot leases if we name it as `SnapshotLeaseChanged` .. if we name it `SnapshotRevisionChanged`, the person needs to understand quite a lot about backup-restore internals.

Reply:
Well if someone is looking at etcd druid code base then first confusion point will be why do you even need a lease for a full and delta snapshot. Second confusing point will be what is this weird looking number mean when i look at the holderIdentity which is meant to hold something completely different. 

So from your own arguments it now becomes very clear to me when the predicate is named as `SnapshotRevisionChanged`. Any code reader would then realize that this predicate will check that the snapshot (full or delta) revision has changed and how it checks is the implementation detail. When he/she looks at how its done then they will come to realize that the holderIdentity holds the revision.

Reply:
Renamed